test/Microsoft.AspNetCore.Localization.Tests/CookieRequestCultureProviderTest.cs

```diff
@@ -117,24 +124,87 @@ public class CookieRequestCultureProviderTest
                             new CultureInfo("ar-SA")
                         }
                     };
-                    var provider = new CookieRequestCultureProvider();
-                    provider.CookieName = "Preferences";
+                    var provider = new CookieRequestCultureProvider
+                    {
+                        CookieName = "Preferences"
+                    };
+                    options.RequestCultureProviders.Insert(0, provider);
+                    app.UseRequestLocalization(options);
+                    app.Run(context =>
+                    {
+                        var requestCultureFeature = context.Features.Get<IRequestCultureFeature>();
+                        var requestCulture = requestCultureFeature.RequestCulture;
+                        Assert.Equal("en-US", requestCulture.Culture.Name);
+                        return Task.FromResult(0);
+                    });
+                });
+
+            using (var server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+                var response = await client.GetAsync(string.Empty);
+            }
+        }
+
+        [Theory]
+        [InlineData("??", "en-US", "CookieRequestCultureProvider returned the following unsupported cultures '??'.")]
+        [InlineData("en-US", "??", "CookieRequestCultureProvider returned the following unsupported cultures '??'.")]
+        public async Task UnsupportedCulturesShouldLogWarning(string culture, string uiCulture, string expectedMessage)
```

Comment:
Let's split these in two separate tests. `RequestLocalizationMiddleware_LogsWarningsForUnsupportedCultures` \  `RequestLocalizationMiddleware_LogsWarningsForUnsupportedUICultures`. This isn't a very good use of `TheoryData`. Looks good otherwise.